src/parser/mod.rs

```diff
@@ -15116,13 +15116,20 @@ impl<'a> Parser<'a> {
         } else {
             None
         };
+        let limit =
+            if self.dialect.supports_update_delete_limit() && self.parse_keyword(Keyword::LIMIT) {
+                Some(self.parse_expr()?)
+            } else {
+                None
+            };
```

Comment:
Since the syntax doesn't seem to conflict with other dialects, can we have the parser accept the clause if it shows up, avoiding the dialect method. e.g.
```rust
let limit = self.maybe_parse(|p| {
  if p.parse_keyword(Keyword::LIMIT) {
    Some(self.parse_expr()?)
  }
  None
})?;
```

Reply:
Yeah that would simplify things. It means that sqlparser can accept a statement and sqlite won't (since `update ... limit ...` is behind a compilation flag). I though it would be better for the user to configure it, but up to you.

Reply:
Yeah I think it would be fine to let the parser always accept the LIMIT clause if it shows up

Reply:
Ok, pushed the changes.